pkg/service/service.go

```diff
@@ -96,7 +98,8 @@ func Apply(
 ) (err error) {
 	logger := log.WithName("service")
 
-	defer func() {
+	// Variable err and ctx are captured by closure.
+	cleanUp := func() {
```

Comment:
`cleanUp` is usually used for resource release and file removal, etc. You can use `setStatus`, `updateStatus` when performing this action.